File: Competitions/UC/Berkeley-2024/contest/contest/celeste/templates/celeste_template.py
def solve(N: int, M: int, K: int, C: list[str]) -> int:
    """
    Return the minimum number of moves to get to E, or return 
    -1 if it is impossible.

    N: number of rows
    M: number of columns
    K: the length of Madeline’s dash
    C: a list of N strings with M characters each, describing 
    the maze Madeline is in.
       In each string:
        . denotes a space
        # denotes a wall.
        * denotes a dash crystal.
        S denotes Madeline’s start tile.
        E denotes Madeline’s end tile.

    """
    # YOUR CODE HERE
    return -1


def main():
    T = int(input())
    for _ in range(T):
        N, M, K = [int(x) for x in input().split()]
        C = [str(input().strip()) for x in range(N)]
        print(solve(N, M, K, C))


if __name__ == '__main__':
    main()

File: Competitions/UC/Berkeley-2024/contest/contest/celeste/templates/celeste_template.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

/**
 * Return the minimum number of moves to get to E, or return 
 * -1 if it is impossible.
 * 
 * N: number of rows
 * M: number of columns
 * K: the length of Madeline’s dash
 * C: a list of N strings with M characters each, describing 
 * the maze Madeline is in.
 *    In each string:
 *     . denotes a space
 *     # denotes a wall.
 *     * denotes a dash crystal.
 *     S denotes Madeline’s start tile.
 *     E denotes Madeline’s end tile.
*/
int solve(int N, int M, int K, vector<string> C) {
    // YOUR CODE HERE
    return -1;
}

int main() {
    int T;
    cin >> T;
    while (T--) {
        int N, M, K;
        cin >> N >> M >> K;
        vector<string> C(N);
        for (int i = 0; i < N; ++i) {
            cin >> C[i];
        }
        cout << solve(N, M, K, C) << '\n';
    }
}
